ci-automation/util/update_cros_workon.sh

```diff
@@ -0,0 +1,111 @@
+#               NOTE that in order for "push" to work the git submodule must be at the tip of a branch.
+#               If in detached head mode, the branch will be identified and checked out before pushing.
+
+function update_cros_workon() {
+    set -euo pipefail
+
+    local ebuild_dir="$1"
+    local branch="${2:-HEAD}"
+    local commit="${3:-false}"
+
+    # Use a subshell and operate directly in $ebuild_dir
+    (
+        cd "${ebuild_dir}"
+
+        # If we're in detached head mode and are supposed to push later then
+        #  check out the branch the current commit is the tip of.
+        # We do this early so we error out before touching anything.
+        if [ "${commit}" = "push" ] ; then
+
+            local ebuild_branch="$(git -C branch --show-current)"
+            if [ "${commit}" = "push" -a -z "${ebuild_branch}" ] ; then
+                # We're in headless mode, see if we can find a branch tip on origin
+                local sha="$(git rev-parse HEAD)"
+                ebuild_branch="$(git ls-remote --heads origin \
+                                | grep "${sha}" \
+                                | sed 's:.*refs/heads/::')"
+            fi
+
+            if [ -z "${ebuild_branch}" ] ; then
+                echo "ERROR updating '${ebuild_dir}': 'push' was requested but current commit '${sha}' is on the tip of any branch!"
+                return 1
+            fi
+
+            # check out the branch the current commit is the tip of.
+            git checkout "${ebuild_branch}"
+        fi
+
+        softlink="$(basename $(find . -type l))"
+
+        name="$(echo "${softlink}" | sed 's/^\(.*\)-[0-9.]\+[-.]*.*\.ebuild/\1/')"
+        version="$(echo "${softlink}" | sed 's/^.*-\([0-9.]\+\)[-.]*.*\.ebuild/\1/')"
+        release="$(echo "${softlink}" | sed -n 's/.*-[0-9.]\+[-.]r\([0-9]\+\)\.ebuild/\1/p')"
+        new_release="$((release + 1))"
+
+        ebuild="${name}-9999.ebuild"
+        new_softlink="${name}-${version}-r${new_release}.ebuild"
+        ln -s "${ebuild}" "${new_softlink}"
+
+        local repo="$(eval "$(grep -E '^\s*(CROS_WORKON_REPO|CROS_WORKON_PROJECT)=' "${ebuild}")"
+                      echo $CROS_WORKON_REPO/$CROS_WORKON_PROJECT)"
+        commit_id="$(git ls-remote "${repo}" "${branch}" | awk "/[[:space:]]${branch}/ {print \$1}")"
+
+        sed -i "s/CROS_WORKON_COMMIT=.*/CROS_WORKON_COMMIT=\"${commit_id}\" # tip of branch ${branch} $(date)/" \
+                "${ebuild}"
+
+        rm "${softlink}"
+
+        echo "Updated '${ebuild}' to commit '${commit_id}'"
+        echo " ('${softlink}' ==> '${new_softlink}'"
+
+        if [[ "${commit}" =~ (commit|push) ]] ; then
+            git add .
+            local category="$(dirname "${PWD}")"
+            category="${category##*/}" # strip everything from the beginning to the last slash
+            git commit -m "${category}/${name}: Update to latest ${branch}"
+
+            if [ "${commit}" = "push" ] ; then
+                git push
```

Comment:
The submodules are configured with https url, so the push will not work because there's no way to pass credentials.